lisa/tools/timeout.py

```diff
@@ -12,20 +12,36 @@ def can_install(self) -> bool:
         return False
 
     def run_with_timeout(
-        self, command: str, timeout: int, signal: int = SIGTERM, kill_timeout: int = 0
+        self,
+        command: str,
+        timeout: int,
+        signal: int = SIGTERM,
+        kill_timeout: int = 0,
     ) -> ExecutableResult:
         # timeout [OPTION] DURATION COMMAND [ARG]...
+
+        return self.start_with_timeout(
+            command=command,
+            timeout=timeout,
+            signal=signal,
+            kill_timeout=kill_timeout,
+        ).wait_result()
+
+    def start_with_timeout(
+        self,
+        command: str,
+        timeout: int,
+        signal: int = SIGTERM,
+        kill_timeout: int = 0,
+        delay_start: int = 0,
+    ) -> Process:
+        # timeout [OPTION] DURATION COMMAND [ARG]...
         params = f"-s {signal} --preserve-status {timeout} {command}"
         if kill_timeout:
             params = f"--kill-after {kill_timeout} " + params
-        command_timeout = timeout
```

Comment:
It looks this logic is missed. It should be in run_with_timeout.

Reply:
apologies, merge issue needs to be resolved. I'll abandon this and re-open when it's ready

Reply:
Is it added back?